### PR TITLE
 Add optional-gui to nsm.h

### PR DESCRIPTION
**Pull request by brummer10**
This add the needed callbacks and methods for optional-gui to nsm.h in order to simplify the inclusion of NSM support

### CODE REVIEW

File: src/nsm.h [hunk 1]
```diff
@@ -26,6 +26,8 @@
 /*                                                           */
 /* #include "nsm.h"                                          */
 /*                                                           */
+/* static nsm_client_t *nsm = 0                              */
```
Comment (falkTX) [inline src/nsm.h:29]: value should be NULL, also missing ; at the end

File: src/nsm.h [hunk 2]
```diff
@@ -45,7 +47,30 @@
...
+/*     do_hide_ui();                                         */
+/*     nsm_send_is_hidden ( nsm );                           */
+/* }                                                         */
+/*                                                           */
+/* gboolean                                                  */
+/* poll_nsm()                                                */
+/* {                                                         */
+/*     if (nsm)                                              */
+/*     {                                                     */
+/*         nsm_check_nowait(nsm);                            */
```
Comment (falkTX) [inline src/nsm.h:69]: a space between the function name and parenthesis is needed here, so it follows the style of the rest of the file

File: src/nsm.h (hunk 1 above)
Comment (falkTX) [reply src/nsm.h:29]: actually, the code here always uses 0 instead of NULL, so ignore that. but still missing ;

Comment (brummer10) [reply src/nsm.h:29]: Yes, you are right. This is pseudo code, and this line is taken directly from the original, means original use 0 and miss the semicolon.

File: src/nsm.h (hunk 2 above)
Comment (brummer10) [reply src/nsm.h:69]: ups, will do.

File: src/nsm.h [hunk 3]
```diff
@@ -57,10 +82,14 @@
 /*                                                           */
 /*         nsm_set_open_callback( nsm, cb_nsm_open, 0 );     */
 /*         nsm_set_save_callback( nsm, cb_nsm_save, 0 );     */
+/*         nsm_set_show_callback( nsm, cb_nsm_show, 0 );     */
+/*         nsm_set_hide_callback( nsm, cb_nsm_hide, 0 );     */
 /*                                                           */
 /*         if ( 0 == nsm_init( nsm, nsm_url ) )              */
 /*         {                                                 */
 /*             nsm_send_announce( nsm, "FOO", "", argv[0] ); */
+/*             nsm_check_wait(nsm, 200);                     */
```
Comment (falkTX) [inline src/nsm.h:91]: why is this added here?

Comment (brummer10) [reply src/nsm.h:91]: This allow the client to wait for the informations needed from the manager, eg, client_id.

**Comment by falkTX** [comment]
Looks good, though I dislike that there is no checks against ":optional-gui:" server-side support and just assumes it is suported.

File: src/nsm.h (hunk 3 above)
Comment (falkTX) [reply src/nsm.h:91]: then please add a comment regarding that.
depending on how an application is made, this is quite optional.
(carla does not need this, for example)

also, this could take more than 200ms, so relying on it is problematic.

File: src/nsm.h (hunk 1 above)
Comment (brummer10) [reply src/nsm.h]: Fixed now.

File: src/nsm.h (hunk 2 above)
Comment (brummer10) [reply src/nsm.h]: Fixed now.

File: src/nsm.h (hunk 3 above)
Comment (brummer10) [reply src/nsm.h]: I've increase the timeout to 100s, add a exit handler and a disclaimer to make clear that this part isn't needed most times.

**Comment by diovudau** [comment]
Thank you very much!
